File: languages/js/src/entity/identity_v2/did_params.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { NetworkId } from "./network";
import { Blockchain } from "./blockchain";
import { Method } from "./method";

/**
 * Represents parameters used for generating DIDs.
 */
export class DidParams {
  method: Method;
  blockchain: Blockchain;
  network: NetworkId;

  /**
   * Constructs a DidParams object with the specified parameters.
   * @param method 
   * @param blockchain 
   * @param network 
   */
  constructor(method: Method, blockchain: Blockchain, network: NetworkId) {
    this.method = method;
    this.blockchain = blockchain;
    this.network = network;
  }

  public toProto(): identityEntitiesProto.DidParams {
    return identityEntitiesProto.DidParams.fromPartial({
      method: Method.toProto(this.method),
      blockchain: Blockchain.toProto(this.blockchain),
      networkId: NetworkId.toProto(this.network)
    });
  }
}

File: languages/python/bloock/entity/identity_v2/did_params.py
import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.method import Method
from bloock.entity.identity_v2.blockchain import Blockchain
from bloock.entity.identity_v2.network import Network


class DidParams:
    """
    Represents parameters used for generating DIDs.
    """
    def __init__(self, method: Method, blockchain: Blockchain, network: Network) -> None:
        """
        Constructs a DidParams object with the specified parameters.
        :type network: object
        :type blockchain: object
        :type method: object
        :rtype: object
        """
        self.method = method
        self.blockchain = blockchain
        self.network = network

    def to_proto(self) -> proto.DidParams:
        return proto.DidParams(
            method=Method.to_proto(self.method),
            blockchain=Blockchain.to_proto(self.blockchain),
            network_id=Network.to_proto(self.network)
        )
